<launch>
    <arg name="bin_path"
         default="$(find rcj_pcms_base)../models/intel/person-reidentification-retail-0277/FP32/person-reidentification-retail-0277.bin"/>
    <arg name="xml_path"
         default="$(find rcj_pcms_base)../models/intel/person-reidentification-retail-0277/FP32/person-reidentification-retail-0277.xml"/>
    <arg name="wait_for_start" default="false"/>
    <node
        name="pf_initializer"
        pkg="person_follower"
        type="PFInitialize.py"
        required="false"
        output="screen"
    >
        <param name="bin_path" type="str" value="$(arg bin_path)"/>
        <param name="xml_path" type="str" value="$(arg xml_path)"/>
        <param name="wait_for_start" type="bool" value="$(arg wait_for_start)"/>
    </node>
    <node
        name="person_follower"
        pkg="person_follower"
        type="PersonFollower.py"
        required="true"
        output="screen"
    >
        <param name="bin_path" type="str" value="$(arg bin_path)"/>
        <param name="xml_path" type="str" value="$(arg xml_path)"/>
    </node>
    <node
        name="pf_visualizer"
        pkg="person_follower"
        type="PFResultShower.py"
        required="true"
        output="screen"
    />
    <node
        name="waypoint_recorder"
        pkg="person_follower"
        type="WayPointRecorder.py"
        required="true"
        output="screen"
    />
</launch>
